runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -1984,6 +2034,103 @@ private void sendWorkerUpdatesToDataflowService(
     }
   }
 
+  private boolean isDistributionCounterUpdate(CounterUpdate counterUpdate) {
+    return counterUpdate.getStructuredNameAndMetadata() != null
```

Comment:
shall we also check `NameAndKind` here.

Reply:
Also we can extract the whole logic to a method which takes a `kind` and return true or false.

Reply:
We need to check both: StructuredNameAndMetadata as well as NameAndKind.

Switching from NameAndKind to StructuredNameAndMetadata was never complete, so we need to support both.

Reply:
done.